components/tidb_query_datatype/src/codec/mysql/enums.rs

```diff
@@ -0,0 +1,101 @@
+use std::cmp::Ordering;
+use tikv_util::buffer_vec::BufferVec;
+
+#[derive(Clone, Debug)]
+pub struct Enum {
+    data: BufferVec,
```

Comment:
Add an `Arc` to reduce clone cost?

Reply:
Yes, I have that plan, but so far I'm not particularly sure what the rest of it will look like, so I'm going to think about optimizing it when it's implemented. I'll just implement the minimum in this PR.

Reply:
Then how about let's removing the `Clone` first, in case of forgetting this optimization?

Reply:
Let me try it out.

Reply:
`ScalarValue` requires `Enum` and `Set` to have `Clone`, maybe I can add a TODO here?

Reply:
Added as https://github.com/tikv/tikv/pull/8849/commits/3dec93af917095ce44d96aa22412d89916ce5b3d